src/systems/collision.rs

```diff
@@ -0,0 +1,180 @@
+                    Collider{transform: standing_transform, bound: standing_bounds})
+                {
+                    info!("Collision {}", collision);
+
+                    collision_resource.0.push(Collision{collider: moving_id, collidee: standing_id, vector: collision});
+                }
+            }
+        }
+    }
+}
+
+pub struct CollisionHandlerForObstacles;
+
+impl<'s> System<'s> for CollisionHandlerForObstacles {
+    type SystemData = (
+        ReadExpect<'s, Collisions>,
+        WriteStorage<'s, Transform>
+        );
+
+    fn run(&mut self, (collisions, mut transforms): Self::SystemData) {
+        for collision in &collisions.0 {
+            if let Some(transform) = transforms.get_mut(collision.collider)
+            {
+                transform.prepend_translation_x(-collision.vector.x);
+                transform.prepend_translation_y(-collision.vector.y);
+            }
+        }
+
+    }
+}
+
+pub struct ProjectileCollisionSystem;
+
+impl<'s> System<'s> for ProjectileCollisionSystem {
+    type SystemData = (
+        WriteStorage<'s, Transform>,
+        ReadStorage<'s, Projectile>,
+        ReadStorage<'s, BoundingCircle>,
+        Entities<'s>,
+        WriteExpect<'s, ProjectileCollisions>
+        );
+    fn run(&mut self, (transforms, projectiles, bounding_circles, entities, mut collision_resource): Self::SystemData) {
+        collision_resource.0.clear();
+        for projectile in (&transforms, &projectiles, &entities).join()
+        {
+            for object in (&transforms, &bounding_circles, &entities).join()
+            {
+                // unlikely
+                if projectile.2 == object.2
+                {
+                    continue;
+                }
+
+                if let Some(collision) = check_projectile_collision(
+                    projectile.0,
+                    Collider{transform: object.0, bound: object.1})
+                {
+                    collision_resource.0.push(ProjectileCollision{
+                        projectile: projectile.2,
+                        target: object.2,
+                        vector: collision});
+                }
+            }
+        }
+    }
+}
+
+pub struct ProjectileCollisionHandler;
+
+impl<'s> System<'s> for ProjectileCollisionHandler {
+    type SystemData = (
+        Entities<'s>,
+        ReadExpect<'s, ProjectileCollisions>
+        );
+
+    // Here Projectile components are not explicitly filtered. ProjectCollisionSystem is expected
+    // to put proper entities in `collision.projectile`
+    fn run(&mut self, (entities, collisions): Self::SystemData) {
+        for collision in &collisions.0 {
+            entities.delete(collision.projectile).expect("JAJAJa");
+        }
+
+    }
+}
+
+struct Collider<'a>
+{
+    transform: &'a Transform,
+    bound: &'a BoundingCircle,
+}
+
+fn calculate_disposition(a: &Transform, b: &Transform) -> Vector2<f32>
+{
+    Vector2::new(
+        b.translation().x - a.translation().x,
```

Comment:
Not sure, but nalgebra should have an operation for subtracting vectors (componentwise subtract).

Maybe this one?
https://docs.rs/nalgebra/0.24.1/nalgebra/base/struct.Matrix.html#impl-Sub%3CMatrix%3CN%2C%20R2%2C%20C2%2C%20SB%3E%3E

Reply:
Here resizing was also desired. I extracted this to `to_vector2` and then I do the subtraction. (I did the resizing "manually" since I didn't find `fixed_resize` for Vector3)